File: android/src/main/java/com/cloudcare/ft/mobile/sdk/tracker/reactnative/FTMobilePackage.java
package com.cloudcare.ft.mobile.sdk.tracker.reactnative;

import androidx.annotation.NonNull;

import com.facebook.react.ReactPackage;
import com.facebook.react.bridge.NativeModule;
import com.facebook.react.bridge.ReactApplicationContext;
import com.facebook.react.uimanager.ViewManager;

import java.util.ArrayList;
import java.util.Collections;
import java.util.List;

public class FTMobilePackage implements ReactPackage {

    @Override
    public List<NativeModule> createNativeModules(ReactApplicationContext reactContext) {
        List<NativeModule> modules = new ArrayList<>();
        modules.add(new FTMobileModule(reactContext));
        modules.add(new FTLogModule(reactContext));
        modules.add(new FTTraceModule(reactContext));
        modules.add(new FTRUMModule(reactContext));
        return modules;
    }

  @NonNull
  @Override
  public List<ViewManager> createViewManagers(@NonNull ReactApplicationContext reactApplicationContext) {
    return Collections.emptyList();
  }


}

File: android/src/main/java/com/cloudcare/ft/mobile/sdk/tracker/reactnative/FTMobilePackage.kt
package com.cloudcare.ft.mobile.sdk.tracker.reactnative

import android.util.Log
import com.facebook.react.ReactPackage
import com.facebook.react.bridge.NativeModule
import com.facebook.react.bridge.ReactApplicationContext
import com.facebook.react.uimanager.ViewManager


class FTMobilePackage : ReactPackage {
  override fun createNativeModules(reactContext: ReactApplicationContext): List<NativeModule> {
    return listOf(
      FTMobileModule(reactContext),
      FTLogModule(reactContext),
      FTTraceModule(reactContext),
      FTRUMModule(reactContext)
    )
  }

  override fun createViewManagers(reactContext: ReactApplicationContext): List<ViewManager<*, *>> {
    return emptyList()
  }
}
